src/tools/rbd_mirror/group_replayer/Replayer.cc

```diff
@@ -754,23 +781,26 @@ void Replayer<I>::try_create_group_snapshot(
       if (!prev_snap_id.empty()) {
         snap = std::next(remote_snap); // attempt to sync next remote snapshot
       }
-      if (snap != m_remote_group_snaps.end() &&
-          snap->state == cls::rbd::GROUP_SNAPSHOT_STATE_COMPLETE) {
+      if (snap != m_remote_group_snaps.end()) {
         auto next_remote_snap_ns = std::get_if<cls::rbd::GroupSnapshotNamespaceMirror>(
             &snap->snapshot_namespace);
         if (next_remote_snap_ns == nullptr) {
-          dout(10) << "found remote user group snapshot: "
-                   << snap->id << dendl;
-          create_group_snapshot(*snap, locker);
-          continue;
+          if(snap->state == cls::rbd::GROUP_SNAPSHOT_STATE_CREATED) {
```

Comment:
even for mirror snapshots `snap->state == cls::rbd::GROUP_SNAPSHOT_STATE_CREATED` ?